### DISCUSSION TITLE
How to remove extra line breaks in ObjectPattern destructured function args?

### DISCUSSION THREAD

**Discussion by charliematters** [prettier/prettier#16452] [Q&A]
We've had some changes made in PRs which have left gaps in our react component function args, and I can't find a good way of removing them with eslint or prettier. Does anyone have any advice?

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAJhACoAWOUA5vgBTAA6U++qEEANEy9-gEYBDAE5coAXwCUwMSA4gIAB1zQAzslDChEAO4AFYQjUoBAG20CAnmrl8hAsAGs4MAMoCAtnAAylOMlSmKnA2do7OLgr2lFTIMELowSBB7jix8YlwAB4KcEI4nrCmJLlQwjhwRgEmQXIq0SZwAIroEPD+gYkAViqZLvVNLW1IVTUgAI6DcLpaCkYgAioAtFBwcIRrsiBxAjgm0QDCEO7uAsjzJiabddQNAIIwcTh86PC6uT4r7dWJZDDuJgB1CjwFSRMBwFyGHC4ABu0IsZzAKmsIBhCQAklB1rAXGA8kpbliXDALA0vqMFFoggC7AozqZzFZNpQgkIYNMBFQTuTEpEhKyziScio8TglJtKZQYACcIQYGRkAAOAAMciEcAmOHVHK5p2GHTkMAEfBlcoVSAATHJ0EESMbKgaQHB3Hw1utCF4BNR0Jy4AAxCBCE4PaL0l4QEBiMRAA

**Comment by gitsult4n** [comment]
This is intentional and there is no Prettier option to strip those blank lines

Prettier preserves empty lines exactly as they were written — from the [rationale docs](https://prettier.io/docs/rationale#empty-lines):

> The approach that Prettier takes is to preserve empty lines the way they were in the original source code.

It only collapses multiple blank lines into one and removes them at the start and end of blocks

Destructured args go through the same printer as object literals and it explicitly keeps a blank line after a property when the source has one — see [print/object.js](https://github.com/prettier/prettier/blob/0bc958e734b00907e2bae2bae45c664ad8a1a2f7/src/language-js/print/object.js#L108-L112)

```js
separatorParts = [",", line];
if (isNextLineEmpty(node, options)) {
  separatorParts.push(hardline);
}
```

So the fix is a one time cleanup — delete the gaps yourself with an editor regex
replace `,\n\s*\n` with `,\n` in the affected files
Once they are gone Prettier will never add them back
